internal/storage/caching/fast_stat_bucket.go

```diff
@@ -279,8 +288,14 @@ func (b *fastStatBucket) DeleteObject(
 }
 
 func (b *fastStatBucket) DeleteFolder(ctx context.Context, folderName string) error {
-	b.invalidate(folderName)
 	err := b.wrapped.DeleteFolder(ctx, folderName)
+	if err != nil {
+		return err
+	}
+	b.invalidate(folderName)
```

Comment:
I feel, you don't have to invalidate since you are overwriting the entry.

Reply:
Makes sense. Removed it.